tools/src/test/java/org/apache/kafka/tools/consumer/group/DescribeConsumerGroupTest.java

```diff
@@ -0,0 +1,840 @@
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+        TestUtils.waitForCondition(() -> {
+            ConsumerGroupCommand.GroupState state = service.collectGroupState(GROUP);
+            return Objects.equals(state.state(), "Stable") && Objects.equals(state.group(), GROUP) && state.numMembers() == 2;
+        }, "Expected a stable group with two members in describe group state result.");
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_NAME)
+    @ValueSource(strings = {"zk", "kraft", "kraft+kip848"})
+    public void testDescribeSimpleConsumerGroup(String quorum) throws Exception {
+        // Ensure that the offsets of consumers which don't use group management are still displayed
+
+        createOffsetsTopic(listenerName(), new Properties());
+        String topic2 = "foo2";
+        createTopic(topic2, 2, 1, new Properties(), listenerName(), new Properties());
+        addSimpleGroupExecutor(Arrays.asList(new TopicPartition(topic2, 0), new TopicPartition(topic2, 1)), GROUP);
+
+        String[] cgcArgs = new String[]{"--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--group", GROUP};
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs);
+
+        TestUtils.waitForCondition(() -> {
+            scala.Tuple2<Option<String>, Option<Seq<ConsumerGroupCommand.PartitionAssignmentState>>> res = service.collectGroupOffsets(GROUP);
+            return res._1.map(s -> s.contains("Empty")).getOrElse(() -> false)
+                && res._2.isDefined() && res._2.get().count(s -> Objects.equals(s.group(), GROUP)) == 2;
+        }, "Expected a stable group with two members in describe group state result.");
+    }
+
+    @ParameterizedTest(name = TEST_WITH_PARAMETERIZED_QUORUM_AND_GROUP_PROTOCOL_NAMES)
+    @MethodSource({"getTestQuorumAndGroupProtocolParametersAll"})
+    public void testDescribeGroupWithShortInitializationTimeout(String quorum, String groupProtocol) {
+        // Let creation of the offsets topic happen during group initialization to ensure that initialization doesn't
+        // complete before the timeout expires
+
+        String[] describeType = DESCRIBE_TYPES[RANDOM.nextInt(DESCRIBE_TYPES.length)];
+        String group = GROUP + String.join("", describeType);
+        // run one consumer in the group consuming from a single-partition topic
+        addConsumerGroupExecutor(1, groupProtocol);
+        // set the group initialization timeout too low for the group to stabilize
+        List<String> cgcArgs = new ArrayList<>(Arrays.asList("--bootstrap-server", bootstrapServers(listenerName()), "--describe", "--timeout", "1", "--group", group));
+        cgcArgs.addAll(Arrays.asList(describeType));
+        ConsumerGroupCommand.ConsumerGroupService service = getConsumerGroupService(cgcArgs.toArray(new String[0]));
+
+        Throwable e = null;
```

Comment:
We can simplify this by following code style.

```java
        ExecutionException e = assertThrows(ExecutionException.class, service::describeGroups);
        assertInstanceOf(TimeoutException.class, e.getCause());
```

Reply:
Done